airbyte-integrations/connectors/destination-gcs/src/test/java/io/airbyte/integrations/destination/gcs/avro/GcsAvroFormatConfigTest.java

```diff
@@ -0,0 +1,167 @@
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.junit.jupiter.api.Test;
+
+class GcsAvroFormatConfigTest {
+
+  @Test
+  public void testParseCodecConfigNull() {
+    List<String> nullConfigs = Lists.newArrayList("{}", "{ \"codec\": \"no compression\" }");
+    for (String nullConfig : nullConfigs) {
+      assertEquals(
+          DataFileConstants.NULL_CODEC,
+          S3AvroFormatConfig.parseCodecConfig(Jsons.deserialize(nullConfig)).toString());
+    }
+  }
+
+  @Test
+  public void testParseCodecConfigDeflate() {
+    // default compression level 0
+    CodecFactory codecFactory1 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"deflate\" }"));
+    assertEquals("deflate-0", codecFactory1.toString());
+
+    // compression level 5
+    CodecFactory codecFactory2 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"deflate\", \"compression_level\": 5 }"));
+    assertEquals("deflate-5", codecFactory2.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigBzip2() {
+    JsonNode bzip2Config = Jsons.deserialize("{ \"codec\": \"bzip2\" }");
+    CodecFactory codecFactory = S3AvroFormatConfig.parseCodecConfig(bzip2Config);
+    assertEquals(DataFileConstants.BZIP2_CODEC, codecFactory.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigXz() {
+    // default compression level 6
+    CodecFactory codecFactory1 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"xz\" }"));
+    assertEquals("xz-6", codecFactory1.toString());
+
+    // compression level 7
+    CodecFactory codecFactory2 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"xz\", \"compression_level\": 7 }"));
+    assertEquals("xz-7", codecFactory2.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigZstandard() {
+    // default compression level 3
+    CodecFactory codecFactory1 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"zstandard\" }"));
+    // There is no way to verify the checksum; all relevant methods are private or protected...
+    assertEquals("zstandard[3]", codecFactory1.toString());
+
+    // compression level 20
+    CodecFactory codecFactory2 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize(
+            "{ \"codec\": \"zstandard\", \"compression_level\": 20, \"include_checksum\": true }"));
+    // There is no way to verify the checksum; all relevant methods are private or protected...
+    assertEquals("zstandard[20]", codecFactory2.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigSnappy() {
+    JsonNode snappyConfig = Jsons.deserialize("{ \"codec\": \"snappy\" }");
+    CodecFactory codecFactory = S3AvroFormatConfig.parseCodecConfig(snappyConfig);
+    assertEquals(DataFileConstants.SNAPPY_CODEC, codecFactory.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigInvalid() {
+    try {
```

Comment:
it's better to use `Assertions.assertThrows(IllegalArgumentException.class, () -> {....} ` instead of `try-catch` block

Reply:
updated, thanks